File: openrest4j-api/src/com/googlecode/openrest/OpenrestException.java
package com.googlecode.openrest;

public class OpenrestException extends Exception {
    public OpenrestException(int httpErrorCode, String error, String errorMessage) {
        super(error + ": " + errorMessage);
        this.httpErrorCode = httpErrorCode;
        this.error = error;
        this.errorMessage = errorMessage;
    }

    public OpenrestException(int httpErrorCode, String error, String errorMessage, Throwable cause) {
        super(error + ": " + errorMessage, cause);
        this.httpErrorCode = httpErrorCode;
        this.error = error;
        this.errorMessage = errorMessage;
    }

    public int httpErrorCode() {
        return httpErrorCode;
    }

    public String error() {
        return error;
    }

    public String errorMessage() {
        return errorMessage;
    }

    private final int httpErrorCode;
    private final String error;
    private final String errorMessage;
}

File: trunk/openrest4net/openrest4net-api/com/googlecode/openrest/OpenrestException.cs
using System;
using System.Collections.Generic;
using System.Net;

namespace com.googlecode.openrest
{
    public class OpenrestException : ApplicationException
    {
        public OpenrestException(HttpStatusCode httpErrorCode, string error, string errorMessage)
        {
            this.httpErrorCode = httpErrorCode;
            this.error = error;
            this.errorMessage = errorMessage;
        }

        public OpenrestException(HttpStatusCode httpErrorCode, string error, string errorMessage, Exception cause)
            : base(cause.Message, cause)
        {
            this.httpErrorCode = httpErrorCode;
            this.error = error;
            this.errorMessage = errorMessage;
        }

        public HttpStatusCode HttpErrorCode
        {
            get
            {
                return httpErrorCode;
            }
        }

        public string Error
        {
            get
            {
                return error;
            }
        }

        public string ErrorMessage
        {
            get
            {
                return errorMessage;
            }
        }

        private readonly HttpStatusCode httpErrorCode;
        private readonly string error;
        private readonly string errorMessage;
    }
}
